FT014 Wyswietlenie widoku wszystkich sekcji

jBachanek-02 opened the pull request.

jBachanek-02 committed:
feat: added section

jBachanek-02 committed:
Merge branch 'develop' into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
feat: modified section

jBachanek-02 committed:
feat: modification sections

jBachanek-02 committed:
feat: downloading data fron ngrx store

jBachanek-02 committed:
feat: downloading data fron ngrx store

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
feat: displayed data from the store

jBachanek-02 committed:
feat: displayed data from the store

jBachanek-02 committed:
fix: removed div from sections.component.html

jBachanek-02 commented:
Widok https://deploy-preview-82--meetek-envelo.netlify.app/sections

piodaw commented on src/app/features/home/home.module.ts:
```diff
@@ -3,6 +3,7 @@ import { RouterModule } from '@angular/router';
 import { StoreModule } from '@ngrx/store';
 import { EffectsModule } from '@ngrx/effects';
 import { HomeComponent } from './home.component';
+import { SectionsComponent } from '../section/sections.component';
 import { SectionEffects } from '../section/store/section.effects';
 import { sectionDetailsReducer, sectionReducer } from '../section/store/section.reducer';
 import { SectionMembersComponent } from '../section/subpages/section-members.component';
```
rozdziel importy wewnętrzne od zewnętrznych

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,38 @@
+          [isActive]="card.isActive"
+          (click)="goToSection()">
+        </app-section-card>
+        <!-- <app-section-card
+          (modification)="card.handleModification()"
+          [name]="'Piłkarze'"
+          [numberOfMembers]="2"
+          [isActive]="false"></app-section-card>
+        <app-section-card [name]="'Rowerzyści'" [numberOfMembers]="24" [isActive]="true"></app-section-card> -->
```
usuń zakomentowane linijki

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,38 @@
+<button mat-fab color="accent" aria-label="Example icon button with a add icon" (click)="goToAddSection()">
+  <mat-icon>add</mat-icon>
+</button>
+<div>
+  {{ allSections$ | async | json }}
+</div>
```
to też nie jest potrzebne

jBachanek-02 replied:
juz usunięte

piodaw commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,102 @@
+  bottom: 5%;
+  right: 5%;
+}
+@media (min-width:1024px) {
+  h1{
+    font-family: var(--roboto);
+    font-weight: normal;
+    padding: 30px;
+    margin-top: 100px;
+    font-size: var(--L_font-size);
+  }
+  .search {
+    display: flex;
+    flex-direction: row;
+    max-width: 310px;
+    position: absolute;
+    top: 10px;
+    right: 5%;
+    width: 90%;
+  }
+  .content{
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    gap: 24px;
+    background-color: var(--surface);
+    width: 500px;
+    box-shadow: 0 5px 4px -2px rgba(0, 0, 0, 0.3);
+    padding: 20px;
+  }
+}
```
korzystaj z globalnych breakpointów

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+  private store = inject(Store);
+  private service = inject(SectionService);
+
+  public readonly allSections$: Observable<Section[]> = this.store.select(selectAllSections);
```
typ i public readonly jest zbędne

jBachanek-02 replied:
ok

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+  @Input() placeholderValue!: string;
+  private router = inject(Router);
+  private store = inject(Store);
+  private service = inject(SectionService);
```
service nie jest używany

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+
+  public readonly allSections$: Observable<Section[]> = this.store.select(selectAllSections);
+
+  public loadSections() {
```
defaultowo wszystkie są public więc dodawanie tego jest zbędne

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+      this.isActive = false;
+    },
+  };
+  goToAddSection() {
+    this.router.navigate([`/${HOME_PATHS.SECTION.ADD}`]);
+  }
+  goToSection() {
+    this.router.navigate([`/${HOME_PATHS.SECTION.SINGLE.CORE}`]);
+  }
```
nawigację dodaj w templatce jako routerLink zamiast na click

jBachanek-02 replied:
ok

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+    handleModification: () => {
+      this.router.navigate([`/${HOME_PATHS.SECTION.SINGLE.SUBPAGES.EDIT}`]);
+    },
+    handleActivation: function () {
+      this.isActive = true;
+    },
+    handleDeactivation: function () {
+      this.isActive = false;
```
logika powinna być połączona z ngrx

piodaw commented on src/app/features/section/shared/list/member-list/member-list.component.ts:
```diff
@@ -30,6 +30,7 @@ export class MembersListComponent implements OnInit {
     this.activeRoute.parent?.paramMap.subscribe(params => {
       const id = params.get('id');
 
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
tego nie powinno być

jBachanek-02 replied:
wyskakuje błąd i nie mogę dac commita

piodaw commented on src/styles.scss:
```diff
@@ -83,6 +83,7 @@ $front-theme: mat.define-dark-theme(
   --disabled: #868686;
   --darkModeBackground: #303030;
   --darkModeSurface: #414148;
+  --text:#000;
```
już jest ta zmienna

jBachanek-02 replied:
rozwiążę konflikt

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+import { MatInputModule } from '@angular/material/input';
+import { MatIconModule } from '@angular/material/icon';
+import { SectionCardComponent, SectionService } from 'src/app/features/section';
+import { HOME_PATHS } from '../home';
```
import powinien być niżej

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+import { NgIf, AsyncPipe, JsonPipe, NgForOf } from '@angular/common';
```
jsonpipe nie jest potrzebny

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,58 @@
+@Component({
+  selector: 'app-sections',
+  standalone: true,
+  imports: [MatButtonModule, MatInputModule, MatIconModule, SectionCardComponent, NgIf, NgForOf, AsyncPipe, JsonPipe],
```
jw

jBachanek-02 committed:
Update member-list.component.ts

jBachanek-02 committed:
fix: optimized code

revised code paying attention on the hints on github

jBachanek-02 committed:
Merge branch 'FT014-feat-wyswietlenie-widoku-wszystkich-sekcji' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,31 @@
+    <ng-container *ngIf="sectionCard as card">
+      <div class="sectionCard" *ngIf="allSections$ | async as sections">
+        <app-section-card
+          routerlink="section/:id"
```
żeby to mogło działać tu musisz przekazać id sekcji

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,31 @@
+    </ng-container>
+  </div>
+</main>
+<a routerLink="section/add">
```
korzystaj ze stałych

jBachanek-02 replied:
zrobiłem na click

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,31 @@
+  </div>
+</main>
+<a routerLink="section/add">
+  <button mat-fab color="accent" aria-label="Example icon button with a add icon">
```
aria-label do wywalenia

jBachanek-02 committed:
feat: added logic from the ngrx store for activating and deactivating sections

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
feat: added search to display sections

and added some changes to search styles

jBachanek-02 committed:
feat: added logic from the ngrx store for activating and deactivating sections

jBachanek-02 committed:
feat: added logic from the ngrx store for activating and deactivating sections

jBachanek-02 committed:
feat: added logic

jBachanek-02 committed:
feat: added logic

jBachanek-02 committed:
feat: added logic

jBachanek-02 committed:
feat: fix warning in commit

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

EMEZYK commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,24 @@
+<main>
```
dodajmy plik index.ts i skróćmy ścieżki importów

jBachanek-02 replied:
Których ?

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+@import 'breakpoints';
+
+main{
```
ogólnie ten komponent wrapujący, powinien być scrollowalny i powinien byc identyczny, jak w przypadku wyświetlania uczestników sekcji; możemy wgl wydzielić sam ten wrapper z uczestników sekcji, do oddzielnego komponentu i tutaj jego użyć, bo w sumie będzie nam potrzebny w kilku miejscach

jBachanek-02 replied:
Możemy
obstawiam że mam to zrobić tak ? XD

EMEZYK replied:
możemy wynieść do oddzielnego taska, aczkolwiek ten będzie zblokowany, dopóki tamten nie będzie zrobiony; więc jak wolisz

jBachanek-02 replied:
XD
Możemy wynieść i sobie go przypiszę XD

jBachanek-02 replied:
i tak muszę to zrobić i tak XD
a przynajmniej jeden task więcej będzie XD

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,62 @@
+    MatIconModule,
+    SectionCardComponent,
+    NgIf,
+    NgForOf,
```
o, nie znam takiej dyrektywki :D ale chyba nawet jej tutaj nie uzywasz, do usuniecia

jBachanek-02 replied:
zmieniłem na NgFor

EMEZYK commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,24 @@
+<main>
+  <h1>Wszystkie sekcje</h1>
+  <div class="content">
+    <br />
```
wydaje mi się, ze ten br jest tu zbędny;  jesli chcesz zwiekszyc odleglosc, to lepiej uzyc margin

EMEZYK commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,24 @@
+        *ngFor="let section of sections"
+        [name]="section.name"
+        (modification)="handleModification(section.id)"
+        (activation)="handleActivation(section.id)"
```
obecnie buttony aktywuj i dezatywuj zmianiają swoją wartość dopiero po odświeżeniu strony

jBachanek-02 replied:
Wiem wiem ale to inny task

jBachanek-02 replied:
jest do aktywacji i dezaktywacji

EMEZYK commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,24 @@
+      <app-section-card
+        *ngFor="let section of sections"
+        [name]="section.name"
+        (modification)="handleModification(section.id)"
```
po kliknięciu w modyfikuj, nie przenosi nas na odpowiedni route

jBachanek-02 replied:
juz tak

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+  gap: 24px;
+}
+h1{
+  font-family: var(--roboto);
```
skoro domyslnie mamy roboto ustawione, to nie musimy tutaj podawac font-family

jBachanek-02 replied:
no w sumie tak zapomniałem

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+  min-height: 100vh;
+  background-color: var(--background);
+}
+.content{
```
tak jak pisalam, skorzystajmy ju z tego wrappera ktory mamy w uczestnikach sekcji

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+h1{
+  font-family: var(--roboto);
+  font-weight: normal;
+  padding: 30px;
```
lepiej dac margin;

ogolnie zwiekszmy tez odleglosc mdz wyszukiwarką a listą tych sekcji w tablecie i mobile

jBachanek-02 replied:
zrobione

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+  bottom: 5%;
+  right: 5%;
+}
+@include breakpoint('laptop') {
```
nie musisz powielac tych wszystkich styli, wystarczy jak dodasz / zmienisz tylko te, ktore roznia sie od tych, ktore ustaliles powyzej

jBachanek-02 replied:
zrobione

EMEZYK commented on src/app/features/section/sections.component.scss:
```diff
@@ -0,0 +1,105 @@
+    box-shadow: none;
+    display: flex;
+    flex-direction: column;
+    padding: 0px !important;
```
unikajmy tych important

jBachanek-02 replied:
oks

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,62 @@
+    this.store.dispatch(SectionActions.deactivateSection({ sectionId: Number(id) }));
+  }
+
+  public loadSections() {
```
public jest domyślne

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,62 @@
+    this.store.dispatch(SectionActions.getSections());
+  }
+
+  ngOnInit() {
```
wyniosłabym wyżej to ngOnInit, pod stora; jak dla mnie bardziej czytelnie jest wtedy, bo wiem co na starcie się odpala; a wszystkie metody sa ponizej

jBachanek-02 committed:
fix: revised the code after reading the comments on github

jBachanek-02 committed:
fix: shortened the import paths

jBachanek-02 committed:
feat: update sections.component

added styles, added mat-list in sections.component.ts and html

jBachanek-02 committed:
feat: added style for mat-list and div.content

jBachanek-02 committed:
feat: modification and activation/deactivation buttons only if i own the section

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,79 @@
+
+import { HOME_PATHS } from '../home';
+import { User } from './shared/interfaces';
+import { SectionCardComponent } from '../section';
+import { selectAllSections } from '../section';
+import { SectionActions } from '../section';
```
można to zapisać w jednym imporcie

piodaw commented on src/app/features/home/home.module.ts:
```diff
@@ -2,10 +2,12 @@ import { NgModule } from '@angular/core';
 import { StoreModule } from '@ngrx/store';
 import { EffectsModule } from '@ngrx/effects';
-import { HomeComponent } from './home.component';
 import { SectionEffects } from '../section/store/section.effects';
 import { sectionDetailsReducer, sectionReducer } from '../section/store/section.reducer';
```
importuj z index.ts

piodaw commented on src/app/features/home/home.module.ts:
```diff
@@ -2,10 +2,12 @@ import { NgModule } from '@angular/core';
 import { sectionDetailsReducer, sectionReducer } from '../section/store/section.reducer';
+
+import { HomeComponent } from './home.component';
 import { SectionMembersComponent } from '../section/subpages/section-members.component';
+import { SectionsComponent } from '../section/sections.component';
```
jw

piodaw commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,79 @@
+    this.store.dispatch(SectionActions.deactivateSection({ sectionId: Number(id) }));
+  }
+
+  loadSections() {
+    this.store.dispatch(SectionActions.getSections());
+  }
```
to można od razu zapisać w ngOnInit a jeśli wolisz tak dodaj private

piodaw commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,36 @@
+      <app-search [placeholderValue]="'Wyszukaj'"></app-search>
+    </div>
+    <mat-list class="list">
+      <ng-container *ngIf="allSections$ | async as sections">
+        <ng-container *ngFor="let section of sections">
```
to można zapisać w jednym ngFor

jBachanek-02 committed:
feat: update sections.component

added sections.component to the index.ts file and shortened imports in home.module

EMEZYK commented on src/app/features/section/sections.component.html:
```diff
@@ -0,0 +1,34 @@
+    <mat-list class="list">
+      <ng-container *ngFor="let section of allSections$ | async as sections">
+        <app-section-card
+          *ngIf="isOwner(section.sectionOwner); else notOwn"
```
na początku przeczytałam noTown xd  może "notOwner" / 'nonOwner"?

jBachanek-02 replied:
zmieniłem na notOwner

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,76 @@
+  private store = inject(Store);
+
+  allSections$ = this.store.select(selectAllSections);
+  loggedUserId!: number;
```
jeśli jest to pole wymagane, to dodaj je do app-sections[]

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,76 @@
+  }
+
+  goToSection(id: number) {
+    this.router.navigate([`/section/${id}`]);
```
podmień te ścieżki w ten sam sposób jak niżej podales addSection

jBachanek-02 replied:
nie działa XD

EMEZYK commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,76 @@
+  }
+
+  modification(id: number) {
+    this.router.navigate([`section/${id}/edit`]);
```
jak niżej

EMEZYK commented on src/app/shared/ui/modals/cancel-confirm-dialog/cancel-confirm-dialog.component.html:
```diff
@@ -8,7 +8,9 @@
   <div class="cancel-confirm-dialog__btn-container">
-    <button (click)="closeDialog()" mat-button class="cancel-confirm-dialog__cancel-btn error">anuluj</button>
-    <button mat-button class="cancel-confirm-dialog__confirm-btn success">potwierdź</button>
+    <button (click)="closeDialog()" mat-button class="cancel-confirm-dialog__cancel-btn error smallButton">
```
o madonno, jaka klasa :D

jBachanek-02 replied:
to nie moja klasa XD

Commit:
feat: changed ng-template name and write [loggedUserId] in selector app.sections

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
fix: corrected the isOwner method

akademiaEnvelo commented on src/app/features/section/sections.component.ts:
```diff
@@ -0,0 +1,79 @@
+  private store = inject(Store);
+
+  allSections$ = this.store.select(selectAllSections);
+  loggedUserId!: number;
```
zamiast wymuszać działanie kompilatora, możemy po prostu przypisać mu wartość inicjalną

```typescript
loggedUserId$ = this.store.select(selectLoggedUser).subscribe(user => {
      if (user) {
        return user.id;
      } else {
       return null;
      }
    })
 ```
 
 i w templatce najpierw
 ```html
 <ng-container *ngIf="loggedUserId$ | async as userId">
    <app-section-card
          *ngIf=userId === section.user.id; else notOwner">
 </ng-container>
  ```

jBachanek-02 committed:
fix: pinned the initial value to the compiler

modified the sections.component.ts and html files

jBachanek-02 committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT014-feat-wyswietlanie-widoku-wszystkich-sekcji

jBachanek-02 committed:
feat: removeed unnecessary imports

jBachanek-02 committed:
fix: changed styles for app-search and add-button in sections.component.scss
